tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -0,0 +1,163 @@
+        assertEquals(set(Arrays.asList(ConsumerGroupState.DEAD, ConsumerGroupState.RECONCILING)), result);
+
+        assertThrows(IllegalArgumentException.class, () -> ConsumerGroupCommand.consumerGroupStatesFromString("bad, wrong"));
+
+        assertThrows(IllegalArgumentException.class, () -> ConsumerGroupCommand.consumerGroupStatesFromString("  bad, Stable"));
+
+        assertThrows(IllegalArgumentException.class, () -> ConsumerGroupCommand.consumerGroupStatesFromString("   ,   ,"));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testListGroupCommand(String quorum) throws Exception {
+        String simpleGroup = "simple-group";
+        addSimpleGroupExecutor(simpleGroup);
+        addConsumerGroupExecutor(1);
+        final String[] out = {""};
+
+        String[] cgcArgs1 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list"};
+        TestUtils.waitForCondition(() -> {
+            out[0] = kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs1);
+                return null;
+            });
+            return !out[0].contains("STATE") && out[0].contains(simpleGroup) && out[0].contains(GROUP);
+        }, "Expected to find " + simpleGroup + ", " + GROUP + " and no header, but found " + out[0]);
+
+        String[] cgcArgs2 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state"};
+        TestUtils.waitForCondition(() -> {
+            out[0] = kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs2);
+                return null;
+            });
+            return out[0].contains("STATE") && out[0].contains(simpleGroup) && out[0].contains(GROUP);
+        }, "Expected to find " + simpleGroup + ", " + GROUP + " and the header, but found " + out[0]);
+
+        String[] cgcArgs3 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "Stable"};
+        TestUtils.waitForCondition(() -> {
+            out[0] = kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs3);
+                return null;
+            });
+            return out[0].contains("STATE") && out[0].contains(GROUP) && out[0].contains("Stable");
+        }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out[0]);
+
+        String[] cgcArgs4 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "stable"};
+        TestUtils.waitForCondition(() -> {
+            out[0] = kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs4);
+                return null;
+            });
+            return out[0].contains("STATE") && out[0].contains(GROUP) && out[0].contains("Stable");
+        }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out[0]);
```

Comment:
Duplicate?

Reply:
No, case of parameter differs - "Stable", "stable".

Reply:
Ah, that's why.